scipy/stats/tests/test_multivariate.py

```diff
@@ -3852,3 +3854,164 @@ def test_broadcasting_moments(self, method_name):
+        assert_equal(res, 0)
+
+        # PDF with out-of-support s2 is not zero if shape parameter is invalid
+        res = stats.normal_inverse_gamma(mu, [-1, np.nan], a, b).pdf(x, -1)
+        assert_equal(res, np.nan)
+
+        res = stats.normal_inverse_gamma(mu, -1, a, b).mean()
+        assert_equal(res, (np.nan, np.nan))
+
+        res = stats.normal_inverse_gamma(mu, lmbda, -1, b).var()
+        assert_equal(res, (np.nan, np.nan))
+
+        with pytest.raises(ValueError, match="Domain error in arguments..."):
+            stats.normal_inverse_gamma(mu, lmbda, a, -1).rvs()
+
+    def test_broadcasting(self):
+        # Test methods with broadcastable array parameters. Roughly speaking, the
+        # shapes should be the broadcasted shapes of all arguments, and the raveled
+        # outputs should be the same as the outputs with raveled inputs.
+        rng = np.random.default_rng(8925849245)
+        b = rng.random(2)
+        a = rng.random((3, 1)) + 2  # for defined moments
+        lmbda = rng.random((4, 1, 1))
+        mu = rng.random((5, 1, 1, 1))
+        s2 = rng.random((6, 1, 1, 1, 1))
+        x = rng.random((7, 1, 1, 1, 1, 1))
+        dist = stats.normal_inverse_gamma(mu, lmbda, a, b)
+
+        # Test PDF and log-PDF
+        broadcasted = np.broadcast_arrays(x, s2, mu, lmbda, a, b)
+        broadcasted_raveled = [np.ravel(arr) for arr in broadcasted]
+
+        res = dist.pdf(x, s2)
+        assert res.shape == broadcasted[0].shape
+        assert_allclose(res.ravel(),
+                        stats.normal_inverse_gamma.pdf(*broadcasted_raveled))
+
+        res = dist.logpdf(x, s2)
+        assert res.shape == broadcasted[0].shape
+        assert_allclose(res.ravel(),
+                        stats.normal_inverse_gamma.logpdf(*broadcasted_raveled))
+
+        # Test moments
+        broadcasted = np.broadcast_arrays(mu, lmbda, a, b)
+        broadcasted_raveled = [np.ravel(arr) for arr in broadcasted]
+
+        res = dist.mean()
+        assert res[0].shape == broadcasted[0].shape
+        assert_allclose((res[0].ravel(), res[1].ravel()),
+                        stats.normal_inverse_gamma.mean(*broadcasted_raveled))
+
+        res = dist.var()
+        assert res[0].shape == broadcasted[0].shape
+        assert_allclose((res[0].ravel(), res[1].ravel()),
+                        stats.normal_inverse_gamma.var(*broadcasted_raveled))
+
+        # Test RVS
+        size = (6, 5, 4, 3, 2)
+        rng = np.random.default_rng(2348923985324)
+        res = dist.rvs(size=size, random_state=rng)
+        rng = np.random.default_rng(2348923985324)
+        shape = 6, 5*4*3*2
+        ref = stats.normal_inverse_gamma.rvs(*broadcasted_raveled, size=shape,
+                                             random_state=rng)
+        assert_allclose((res[0].reshape(shape), res[1].reshape(shape)), ref)
+
+    @pytest.mark.slow
+    @pytest.mark.fail_slow(10)
+    def test_moments(self):
```

Comment:
Could make this xslow (it takes about a second), but it is currently the only test of mean and variance.